google/cloud/aiplatform/experimental/vertex_model/base.py

```diff
@@ -151,15 +151,19 @@ def f(*args, **kwargs):
             with open(script_path, "w") as f:
                 f.write(source)
 
+            # Handle dependencies by first checking for user requirements
+            if not obj._dependencies:
+                obj._dependencies = [
```

Comment:
I think we can keep our default dependencies as a class attribute on the base class.

```
class VertexModel:
    _dependencies = (...,)
```

And allow the user to override. That way we don't need to perform this check.

It also makes the the default dependencies more transparent to the user.